app/components/subject-viewer.cjsx

```diff
@@ -113,11 +114,13 @@ module.exports = React.createClass
             </span>
         </span>}
         <span>
+          {if @props.workflow.configuration.enable_subject_flags
+            <FlagSubjectButton project={@props.project} subject={@props.subject} user={@props.user} workflow={@props.workflow} classification={@props.classification} />}
           {if @props.subject?.metadata?
             <button type="button" aria-label="Metadata" title="Metadata" className="metadata-toggle" onClick={@showMetadata}><i className="fa fa-info-circle fa-fw"></i></button>}
           {if @props.subject? and @props.user? and @props.project?
             <span>
-              <FavoritesButton project={@props.project} subject={@props.subject} user={@props.user} />
+              {<FavoritesButton project={@props.project} subject={@props.subject} user={@props.user} /> unless @props.workflow.disable_favorites }
```

Comment:
Awaiting separate PR for this part. Also, might wanna follow the convention of splitting this conditional into two lines.